--- a/source3/rpc_server/rpc_ep_setup.c
+++ b/source3/rpc_server/rpc_ep_setup.c
@@ -1056,6 +1060,10 @@ bool dcesrv_ep_setup(struct tevent_conte
					   "rpc_server",
					   "epmapper",
					   "none");
+#ifndef PRINTER_SUPPORT
+	if (1) {
+	} else
+#endif
	if (StrCaseCmp(rpcsrv_type, "embedded") == 0) {
		epmapper_cb.init         = epmapper_init_cb;
		epmapper_cb.shutdown     = epmapper_shutdown_cb;

